pennylane/devices/experimental/default_qubit_2.py

```diff
@@ -197,7 +221,19 @@ def execute(
             self.tracker.update(batches=1, executions=len(circuits))
             self.tracker.record()
 
-        results = tuple(simulate(c, rng=self._rng, debugger=self._debugger) for c in circuits)
+        max_workers = self._get_max_workers(execution_config)
+        if max_workers is None:
+            results = tuple(simulate(c, rng=self._rng, debugger=self._debugger) for c in circuits)
+        else:
+            vanilla_circuits = [convert_to_numpy_parameters(c) for c in circuits]
+            _wrap_simulate = partial(simulate, rng=self._rng, debugger=self._debugger)
```

Comment:
Just a comment about the sampling behavior.

```
>>> dev = qml.devices.experimental.DefaultQubit2(seed=42, max_workers=2)
>>> tape = qml.tape.QuantumScript([qml.Hadamard(0)],[qml.sample(wires=0)], shots=9)
>>> dev.execute([tape, tape])
(array([ True, False,  True,  True, False,  True,  True,  True, False]),
 array([ True, False,  True,  True, False,  True,  True,  True, False]))
```

Basically, they are perfectly correlated, because they start sampling from the same rng state.

One solution to this is the fact that the `rng` keyword to `simulate` can also be a seed instead. So instead of sending it off to a new thread with an rng object, we could send it off with a random number pulled from the devices rng.

```
seeds = self._rng.integers(10000000, size=len(vanilla_circuits))
with concurrent.futures.ProcessPoolExecutor(max_workers=max_workers) as executor:
    exec_map = executor.map(simulate, vanilla_circuits, seeds)
```


Though it looks like once we can use `numpy v1.25`, we can use `rng.spawn`: https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.spawn.html#numpy.random.Generator.spawn

Reply:
> One solution to this is the fact that the rng keyword to simulate can also be a seed instead. So instead of sending it off to a new thread with an rng object, we could send it off with a random number pulled from the devices rng.

Good to know, thanks. I'll make the changes.